tracer/src/Datadog.Trace/RemoteConfigurationManagement/Subscription.cs

```diff
@@ -7,21 +7,28 @@
 
 using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace Datadog.Trace.RemoteConfigurationManagement
 {
     internal class Subscription : ISubscription
     {
         private readonly HashSet<string> _productKeys;
 
-        public Subscription(Func<Dictionary<string, List<RemoteConfiguration>>, Dictionary<string, List<RemoteConfigurationPath>>?, IEnumerable<ApplyDetails>> callback, params string[] productKeys)
+        public Subscription(Func<Dictionary<string, List<RemoteConfiguration>>, Dictionary<string, List<RemoteConfigurationPath>>?, ApplyDetails[]> callback, params string[] productKeys)
+        {
+            _productKeys = new HashSet<string>(productKeys);
+            Invoke = (configsByProduct, removedConfigsByProduct) => Task.FromResult(callback(configsByProduct, removedConfigsByProduct));
```

Comment:
Should we/do we need to handle if callback throws an exception with `Task.FromException()`? 🤔

Reply:
The call is awaited in a try/catch so that won't make a difference. You could make a point that methods that return a task shouldn't throw, personally I'm fine either way. Your call.

Reply:
meh, happy to defer to you on this one. As long as nothing terrible happens when we inevitably throw, I'm happy enough